packages/security_center/lib/disk_encryption/disk_encryption_page.dart

```diff
@@ -47,126 +47,378 @@ class EncryptionPageBody extends ConsumerWidget {
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final l10n = AppLocalizations.of(context);
+
+    // Listen for Add PIN/Passphrase success, invalidate state
+    ref.listen<ChangeAuthModeDialogModelData>(
+      changeAuthModeDialogModelProvider(AuthMode.pin),
+      (previous, next) {
+        if (next.dialogState is ChangeAuthModeDialogStateSuccess) {
+          ref.invalidate(tpmAuthenticationModelProvider);
+        }
+      },
+    );
+    ref.listen<ChangeAuthModeDialogModelData>(
+      changeAuthModeDialogModelProvider(AuthMode.passphrase),
+      (previous, next) {
+        if (next.dialogState is ChangeAuthModeDialogStateSuccess) {
+          ref.invalidate(tpmAuthenticationModelProvider);
+        }
+      },
+    );
+
+    // Listen for Remove PIN/Passphrase success, invalidate state
+    ref.listen<RemoveAuthState>(
+      removeAuthModelProvider(AuthMode.pin),
+      (previous, next) {
+        if (previous is RemoveAuthStateLoading && next is RemoveAuthStateIdle) {
+          ref.invalidate(tpmAuthenticationModelProvider);
+        }
+      },
+    );
+    ref.listen<RemoveAuthState>(
+      removeAuthModelProvider(AuthMode.passphrase),
+      (previous, next) {
+        if (previous is RemoveAuthStateLoading && next is RemoveAuthStateIdle) {
+          ref.invalidate(tpmAuthenticationModelProvider);
+        }
+      },
+    );
+
```

Comment:
Can't we just add `ref.invalidate(tpmAuthenticationModelProvider)` to `removeAuth()` in the model?